docs/build/isc/v1.1/docs/tutorials/defi-lend-borrow-tutorial-part-1.md

```diff
@@ -0,0 +1,264 @@
+```
+
+- `preBorrowChecks(address iTokenAddress, address redeemer, uint256 amount)`: Ensures that the redeemer has sufficient collateral to borrow tokens.
+
+- **Parameters:**
+  - `iTokenAddress`: The address of the `IToken` to be borrowed.
+  - `redeemer`: The account attempting to borrow.
+  - `amount`: The amount of tokens to be borrowed.
+- **Reverts:** `BorrowAmountTooMuch` if the collateral is insufficient.
+
+```javascript reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/ITokenManager.sol#L156-L185
+```
+
+- `hasLiquidity(address account, address iToken, uint256 redeemTokens, uint256 borrowTokens) → (uint256 totalAccountCollaterals, uint256 totalAccountBorrows)`: Calculates the total collateral and borrow balances for an account, accounting for tokens to be redeemed or borrowed.
+- **Parameters:**
+  - `account`: The account to check.
+  - `iToken`: The token in question.
+  - `redeemTokens`: The number of tokens to redeem.
+  - `borrowTokens`: The number of tokens to borrow.
+- **Returns:** 
+  - `totalAccountCollaterals`: The total collateral balance.
+  - `totalAccountBorrows`: The total borrow balance.
+
+```javascript reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/ITokenManager.sol#L187-L247
+```
+- `updateTokenUSDPrice(address token, uint256 newUSDPrice)`: Updates the USD price of a supported token.
+
+- **Parameters:**
+  - `token`: The address of the token.
+  - `newUSDPrice`: The new USD price for the token.
+- **Modifiers:** `onlyOwner`, `onlySupportedToken`
+- **Reverts:** `TokenNotListed` if the token is not supported.
+
+```javascript reference
+https://github.com/iota-community/Defi-lend-borrow/blob/bfb658e82611657a50885a2509f21a07da2bbef1/contracts/ITokenManager.sol#L245-L258
+```
+
+- `getTokenUSDPrice(address token) → uint256`: Returns the USD price of a supported token.
+
+- **Parameters:**
+  - `token`: The address of the token.
+- **Returns:** The USD price of the token as a `uint256`.
+
+```javascript reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/ITokenManager.sol#L249-L262
+```
+
+We have now covered all relevant parts and working of the iTokenManager contract, here is the [link](https://github.com/iota-community/Defi-lend-borrow/blob/main/contracts/ITokenManager.sol) for the full contract code for your reference.
+
+### Underlying Token
+
+The `MockERC20` contract is a simple ERC20 token used for testing purposes. It allows minting of an initial supply to the deployer. You can add this token to the `ITokenManager` and mint new ITokens by depositing the MockERC20 tokens as the underlying Token.
+
+
+## Scripts
+
+### Compile Your Contracts
+
+```bash
+npx hardhat compile
+```
```

Comment:
I think this step may work better in the previous section, as it refers to contracts. If not, please add a small explanation .